packages/edit-site/src/store/reducer.js

```diff
@@ -169,24 +158,32 @@ export function page( state = {}, action ) {
 }
 
 /**
- * Reducer returning the site's `show_on_front` setting.
+ * Reducer for information about the site's homepage.
  *
  * @param {Object} state Current state.
+ * @param {Object} action Dispatched action.
  *
  * @return {Object} Updated state.
  */
-export function showOnFront( state ) {
+export function home( state = {}, action ) {
+	switch ( action.type ) {
+		case 'SET_HOME_TEMPLATE':
```

Comment:
In a similar vein as https://github.com/WordPress/gutenberg/pull/23775/files#r451417455 -- a `home` reducer (for setting the home page template), and corresponding `SET_HOME_TEMPLATE` is quite specific.

While it certainly is information that's quite relevant for the site editor, it raises the question why we aren't storing the same information for other pages (or more generally, categories etc). I think we made a decision to not store a lot of template/template parts related information in a dedicated state subtree, but use `getEntitiesRecord` and friends :thinking:

Reply:
homepage data isn't stored anywhere in gutenberg yet, and we need a way to know what the homepage is while looking at other pages on the site. (e.g. to show the "home" icon" beside the homepage in the page switcher)